PopPUNK/utils.py

```diff
@@ -230,25 +237,38 @@ def qcDistMat(distMat, refList, queryList, a_max):
             Reference labels
         queryList (list)
             Query labels (or refList if self)
+        c_max (float)
+            Maximum core distance to allow
         a_max (float)
             Maximum accessory distance to allow
+        ref_isolate (str)
+            Name of reference from which pruning can occur
 
     Returns:
         passed (bool)
             False if any samples failed
     """
     passed = True
+    to_prune = []
 
     # First check with numpy, which is quicker than iterating over everything
-    if np.any(distMat[:,1] > a_max):
+    if np.any(distMat[:,1] > a_max) or np.any(distMat[:,0] > c_max):
         passed = False
         names = iterDistRows(refList, queryList, refList == queryList)
         for i, (ref, query) in enumerate(names):
-            if distMat[i,1] > a_max:
-                sys.stderr.write("WARNING: Accessory outlier at a=" + str(distMat[i,1]) +
+            if distMat[i,0] > c_max or distMat[i,1] > a_max:
+                sys.stderr.write("WARNING: Outlier at c = " + str(distMat[i,0]) + " a = " + str(distMat[i,1]) +
                                  " 1:" + ref + " 2:" + query + "\n")
-
-    return passed
+                if ref_isolate is not None:
+                    if ref == ref_isolate:
+                        to_prune.append(query)
+                    elif query == ref_isolate:
+                        to_prune.append(ref)
+
+    if ref_isolate is None:
+        return passed
+    else:
+        return to_prune
```

Comment:
Single return function ideally.
It's also confusing that this return changes greatly depending on an argument (this isn't reflected in the docstring above either). Is it possible to better harmonise this?

Reply:
Done in 46a9a9a.